File: Searching/count1.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    vector<int> arr{1, 1, 1, 0, 0, 0, 0};
    int left = 0, right = arr.size() - 1;
    int target = 1;
    while(left <= right){
        int mid = left + (right - left) / 2;
        if (arr[mid] == target){
            left = mid + 1;
        }else{
            right = mid - 1;
        }
    }
    cout << left;
}

File: Searching/count1.go
package main

import "fmt"

func main() {
	arr := []int{1, 1, 1, 0, 0, 0, 0}
	left, right := 0, len(arr)-1
	target := 1
	for left <= right {
		mid := left + (right-left)/2
		if arr[mid] == target {
			left = mid + 1
		} else {
			right = mid - 1
		}
	}
	fmt.Println(left)
}
